app/main/routes.py

```diff
@@ -193,10 +198,49 @@ def browse():
         filters=browse_parameters,
         browse_type=browse_type,
         results=browse_results,
+        transferring_bodies=transferring_bodies,
+        user_filters=filters,
         num_records_found=num_records_found,
     )
 
 
+def _build_browse_all_filters():
+    if request.args:
+        filter_items = []
+        filters = {}
+        transferring_body = request.args.get(
+            "transferring_body_filter", ""
+        ).lower()
+        series = request.args.get("series_filter", "").lower()
+        date_from_day = request.args.get("date_from_day", "")
+        date_from_month = request.args.get("date_from_month", "")
+        date_from_year = request.args.get("date_from_year", "")
+        date_from = date_from_day + "/" + date_from_month + "/" + date_from_year
```

Comment:
I was a little confused about the "//" checks below, then realised they are accounting for the possibility of these string concatenations resulting in "//"

What if someone puts:
- day, month, BUT no year
- month, year BUT no day
- etc.

I think we should probably set defaults for the Day, month or year, depending on the other values specified already. I remember us doing something similar in caselaw... https://github.com/nationalarchives/ds-caselaw-public-ui/blob/7005a0ab1e52b6143951f83b3c2887e7a448434e/judgments/views/advanced_search.py#L32

It could be worth creating a `_build_date_range_filter()` to abstract out all this date logic into one place to.

Instead of checking for "??" below, could we just

Reply:
we should be checking the frontend for basic date validation and highlighting any error to the user to amend it when clicking on apply a filter , alternatively,
e.g. if the user enters only year as 2023 - the date from should default to 01/01/2023, the same way if the user enters the only year 2023 - the date to should default to 31/12/2023. and if not data entered for date to, it should default to the current date, this can be implemented, but we might need to spend more time putting the functionality - this is based on a discussion we had today with Terry

Reply:
at the moment separated date filters in _build_date_range_filter , but we need to finalise and agree on what approach we want to take for date parameters

1. we can add validation on front end to check date filters have a valid date value using flaskform
2. we need to create separate ticket for validations

Reply:
![image](https://github.com/nationalarchives/da-ayr-beta-webapp/assets/107114665/bea5d760-5c0f-46f2-a9fa-b85c9ca3c761)

Reply:
we need to provide the validations according to Terry's design
https://national-archives.atlassian.net/wiki/spaces/DAYR/pages/204275723/Validation+messaging+matrix+for+AYR